tlf/handle.go

```diff
@@ -326,3 +345,72 @@ func (h Handle) IsFinal() bool {
 func (h Handle) IsConflict() bool {
 	return h.ConflictInfo != nil
 }
+
+// DeepEqual returns true if the handle is equal to another handle.
+// This can mutate the Handle in that it might sort its Writers,
+// Readers, UnresolvedWriters, and UnresolvedReaders.
+func (h *Handle) DeepEqual(other Handle) bool {
+	if len(h.Writers) != len(other.Writers) {
+		return false
+	}
+	if len(h.UnresolvedWriters) != len(other.UnresolvedWriters) {
+		return false
+	}
+	if len(h.Readers) != len(other.Readers) {
+		return false
+	}
+	if len(h.UnresolvedReaders) != len(other.UnresolvedReaders) {
+		return false
+	}
+
+	if !h.sorted {
+		sort.Sort(UIDList(h.Writers))
+		sort.Sort(UIDList(h.Readers))
+		sort.Sort(SocialAssertionList(h.UnresolvedWriters))
+		sort.Sort(SocialAssertionList(h.UnresolvedReaders))
+		h.sorted = true
+	}
+	if !other.sorted {
+		sort.Sort(UIDList(other.Writers))
+		sort.Sort(UIDList(other.Readers))
+		sort.Sort(SocialAssertionList(other.UnresolvedWriters))
+		sort.Sort(SocialAssertionList(other.UnresolvedReaders))
```

Comment:
Is it intended not to set `sorted` for `other`?

Reply:
Yeah, I want to be able to accept a non pointer comparer. Also, it seems wrong to mutate the comparison target.